crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -1140,11 +1165,15 @@ impl RoomInfo {
         self.summary.invited_member_count = count;
     }
 
-    /// Updates the heroes user ids.
+    /// Updates the room heroes.
     #[cfg(feature = "experimental-sliding-sync")]
-    pub(crate) fn update_heroes(&mut self, heroes: Vec<OwnedUserId>, names: Vec<String>) {
-        self.summary.heroes_user_ids = heroes;
-        self.summary.heroes_names = names;
+    pub(crate) fn update_heroes(&mut self, heroes: Vec<RoomHero>) {
+        self.summary.room_heroes = heroes;
+    }
+
+    /// The heroes for this room.
+    pub fn heroes(&self) -> &[RoomHero] {
+        self.summary.room_heroes.deref()
```

Comment:
this smells like AI-generated code now :stuck_out_tongue: 

```suggestion
        &self.summary.room_heroes
```

Reply:
Ha, although whilst there's definitely an healthy amount of Copilot assistance in my PR, I'm pretty sure I wrote this one myself - I very much forget to `&thing` now and then look for a method to get the ref 🙈